### ISSUE TITLE
Feature Request: warn about extra shapekeys

### ISSUE THREAD

**Issue by manavortex** [WolvenKit/Cyberpunk-Blender-add-on#47]
I had a user who followed the Netrunner refitting tutorial, and at some point, they "used sculpt" to smooth out their mesh edits. The tutorial they followed ended them up with this:

![image](https://github.com/WolvenKit/Cyberpunk-Blender-add-on/assets/965785/1a3d4916-746c-447b-8c67-15d09146dce8)
Naturally, this didn't import, and equally naturally, they had no idea what had even gone wrong.

It would be great if the plugin could check for the existence of any shapekeys that are not Basis and GarmentSupport, and either offer to apply them before export (if possible without fucking up something else), or link to [this wiki page](https://wiki.redmodding.org/cyberpunk-2077-modding/for-mod-creators/3d-modelling/troubleshooting-your-mesh-edits#do-you-have-extra-shapekeys).

**Comment by DoctorPresto** [comment]
The problem with this would be when editing anything with the character morphtargets. There's too many of them to include explicit checks by name and so what you're suggesting would mean that any morphtargets edits that are done intentionally are going to either not work because it applies the morph, or throw a bunch of extra warnings that aren't necessary. I'll try to think through this and come up with something that can catch this as an error but also doesn't have a negative effect on other workflows.

**Comment by effs** [comment]
You absolutely can’t just apply shapekeys, and I really don’t want warnings or opt outs I need to click through.

There’s room for improvement in the error messages, because I wrote them with the assumption that you’re *trying* to import targets. However, even that won’t catch the cases where they change it so that it does import but won’t work expectedly.

Best options I see here are

1. Trying to finally put together some broader guidance on how things the puzzle is assembled

2. Script the specific workflows so that they can check specific things in those specific conditions

**Comment by manavortex** [comment]
Yeah, I was  just thinking about clothing here, not about morphtargets, my bad.
Could the plugin tell that apart, maybe? Tracking if the import was a .morphtarget, and print warnings if otherwise?

**Comment by DoctorPresto** [comment]
That's an option to consider for sure. There's some other things in the works that will require more extensive tagging/identification of what's being imported anyways so I'm sure I can come up with something that works for everyone here. I'll figure it out!

**Comment by DoctorPresto** [comment]
@manavortex looks like we can tag shapekeys with a custom property on import, something like is imported. On export if shapekeys are found, we can check for this property and if it's not found display a warning. We'll need to provide a mechanism for people to add this property to shapekeys they create intentionally. Does that work?

**Comment by manavortex** [comment]
Absolutely!

How much work would it be to ask the user if they want shapekeys to be applied or imported? Could you show something like

We found the following shapekeys that weren't there when you imported the file. Which of them do you want to apply now? (Wolvenkit will ignore shapekeys that it doesn't know)
... and then a list of checkboxesß

Maybe filter out Base and GarmentSupport, since those will always be allowed
